src/lib.rs

```diff
@@ -106,16 +107,166 @@ impl Build {
     }
 }
 
+#[derive(Debug, Clone, StructOpt)]
+pub struct Watch {
+    #[structopt(long = "watch", short = "w")]
+    watch_paths: Vec<PathBuf>,
+    #[structopt(long = "ignore", short = "i")]
+    exclude_paths: Vec<PathBuf>,
+    #[structopt(skip)]
+    workspace_exclude_paths: Vec<PathBuf>,
+}
+
+impl Watch {
+    pub fn new() -> Self {
+        Self {
+            exclude_paths: Vec::new(),
+            watch_paths: Vec::new(),
+            workspace_exclude_paths: Vec::new(),
+        }
+    }
+
+    pub fn exclude(&mut self, path: impl AsRef<Path>) {
+        self.exclude_paths.push(path.as_ref().to_path_buf())
+    }
+
+    pub fn excludes(&mut self, paths: Vec<impl AsRef<Path>>) {
+        for path in paths {
+            self.exclude(path)
+        }
+    }
+
+    pub fn workspace_exclude(&mut self, path: impl AsRef<Path>) {
+        let metadata = cargo_metadata::MetadataCommand::new()
+            .exec()
+            .expect("cannot get workspace metadata");
+
+        self.workspace_exclude_paths
+            .push(metadata.workspace_root.as_std_path().join(path))
+    }
+
+    pub fn workspace_excludes(&mut self, paths: Vec<impl AsRef<Path>>) {
+        for path in paths {
+            self.workspace_exclude(path)
+        }
+    }
+
+    pub fn watch(&mut self, path: impl AsRef<Path>) {
+        self.watch_paths.push(path.as_ref().to_path_buf())
+    }
+
+    pub fn watchs(&mut self, paths: Vec<impl AsRef<Path>>) {
+        for path in paths {
+            self.watch(path)
+        }
+    }
```

Comment:
The naming of those functions is really not great... maybe we should check around how they call that in other libraries.

Reply:
Maybe `exclude_path`, `exclude_paths`, `watch_path`, `watch_paths` and `exclude_workspace_path`, `exclude_workspace_paths` (see the inversion here). Wdyt?

Reply:
True, fixed